### PR TITLE
Web-Console: Adds edit-context-dialog

### PR DESCRIPTION
**Pull request by mcbrewster**
localhost_18081_unified-console html (22)

### Description

Adds an additional MenuItem to the run button that opens the edit-context-modal where the user can modify the context object. The submit button will be greyed out if not valid JSON. When the user submits the object the corresponding checkboxes will also update. 
<hr>

This PR has:
- [x ] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/incubator-druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

Check the items by putting "x" in the brackets for the done things. Not all of these items apply to every PR. Remove the
items which are not done or not relevant to the PR. None of the items from the checklist above are strictly necessary,
but it would be very helpful if you at least self-review the PR.

<hr>

For reviewers: the key changed/added classes in this PR are `MyFoo`, `OurBar`, and `TheirBaz`.

(Add this section in big PRs to ease navigation in them for reviewers.)

### CODE REVIEW

File: web-console/src/views/query-view/run-button/run-button.tsx [hunk 1]
```diff
@@ -133,6 +140,11 @@ export class RunButton extends React.PureComponent<RunButtonProps, RunButtonStat
             onQueryContextChange(setUseCache(queryContext, !useCache));
           }}
         />
+        <MenuItem
+          icon={IconNames.PROPERTIES}
```
Comment (mcbrewster) [inline web-console/src/views/query-view/run-button/run-button.tsx]: please let me know if there is a cooler icon

Comment (vogievetsky) [reply web-console/src/views/query-view/run-button/run-button.tsx]: Good choice!

File: web-console/src/views/query-view/run-button/run-button.tsx [hunk 2]
```diff
@@ -133,6 +140,11 @@ export class RunButton extends React.PureComponent<RunButtonProps, RunButtonStat
...
+          text="Edit Context"
```
Comment (vogievetsky) [inline web-console/src/views/query-view/run-button/run-button.tsx]: lowercase that 'C'

**Comment by vogievetsky** [comment]
Looks good other than that 'C' (not a merge blocker IMO) 👍

**Pull request merged**